hutch_python/user_load.py

```diff
@@ -50,3 +56,181 @@ def get_user_objs(load, *, ask_on_failure=True):
                 )
 
     return objs
+
+
+def update_objs(
```

Comment:
Random question unrelated to this specific line:

Could I use the object configuration from my device class implementation (i.e., from the pcdsdevices side of things)? That is to say, if I want to store user-configurable data for MyDevice, would it fit into this scheme?

Reply:
If the question is: "can I use this configuration framework to set attributes?", I imagine the answer is yes.  We'd just have to write an `update_data` function that accesses the device right?

Reply:
I wasn't sure if it was built in here already without re-reading the source - so I think that is my question, then.

If we all think it's a good thing to have, the next step would be: what would be a good interface to do that? A `BaseInterface` property that does something like `device.conf_yml_data = settings`? A method `set_conf_yml_data` (named differently or made semi-private)?

Reply:
We only work on a per-class basis here, correct? I think my configuration thing may be instance-level at times. Not sure how that could/should be built-in. (@ZLLentz any thoughts on this thread?)

Reply:
The idea you're discussing here is: an API that a class can follow to signify to hutch-python that a class/instance setting can be configurable via this config file, rather than needing to define the specifics externally in `hutch-python`? Is that correct?

First thought: we could support something like that with explicit entrypoint loading or duck-typed methods- hutch-python could see that an object or class has an unknown definition in the config file and check the class for how to handle it. This would probably work reasonably well. I think this second one is what you're suggesting a few comments above?

Second thought: for things that are truly just setattrs... could there be a default handler? A lot of class and object configurations don't need to involve running special code, just class.my_setting = 3

More thought needed maybe to consider the merits of different approaches here

Reply:
I think there are a lot of paths forward here - and it probably doesn't need to be resolved here. Seems like I should open an issue to summarize our suggested approaches and see what we can do about fitting it into the current scheme (since we seem to agree it could be useful).